topics/dev/tutorials/debugging/tutorial.md

```diff
@@ -0,0 +1,730 @@
+>    > (See [Galaxy's contributor's guide](https://github.com/galaxyproject/galaxy/blob/dev/CONTRIBUTING.md#how-to-contribute) for details on how to write a useful and properly formatted commit message.)
+>    >  
+>    >  Awesome work - you now understand how to solve client linting issues!- you now have a handle on client tests!
+{: .hands_on}
+
+# Selenium test failure
+
+Selenium is the end-to-end or integration testing framework that we use; so these tests often are designed to test both frontend and backend code. This makes fixing them a little difficult, but we are sure you can do it!
+
+> ### {% icon hands_on %} Hands-on: Fixing a failing Selenium test
+> 
+> 1. __Finding the failing test on GitHub__
+>    >  
+>    >  There may be actually a few failing Selenium tests, but the good new is, they are the result of only one little error. The failing tests on GitHub says "Selenium Test 3.7, 2". When you click on the details of this one, it's not always very clear which test is causing the failure, but if you download the artifacts, it's very clear which tests have failed.
+>    >  
+>    >  Try to identify which tests are failing based on artifacts you downloaded.
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The failing tests are:
+>    >  >
+>    >  > - `test_build_list_and_show_items`
+>    >  > - `test_data_column_input`
+>    >  > - `test_upload_pair_specify_extension`
+>    >  {: .solution }
+>    >  
+> 2. __Running the tests locally__
+>    >  
+>    >  We do not want to run the full suite of Selenium tests here - that would take a lot of time and we already know that most of them passed.
+>    >  
+>    >  Instead, we can build our commands to only run a subset of the tests. Here's an example: `./run_tests.sh -selenium lib/galaxy_test/selenium/test_sign_out.py`
+>    >  
+>    >  A command like that will still run all of the tests defined in that python file, so we need to find out where the tests that failed are defined. Try seaching the baseline for our three failing Selenium tests; what files are they in?
+>    >  
+>    >  > ### {% icon solution %} Solution
+>    >  >
+>    >  > The failing tests are defined in these files:
+>    >  >
+>    >  > - `lib/galaxy_test/selenium/test_collection_builders.py`
+>    >  > - `lib/galaxy_test/selenium/test_workflow_editor.py`
+>    >  > - `lib/galaxy_test/selenium/test_uploads.py`
+>    >  {: .solution }
+>    >  
+>    >  Now that we have the files, we can run the selenium tests. The command to do this should be very similar to `./run_tests.sh -selenium lib/galaxy_test/selenium/test_sign_out.py`. The only difference should be the python file where the tests are defined. 
+>    >  
```

Comment:
@ic4f, I never give them a sample command to run in a Solution block... I'm certain that quick readers will simply think test_sign_out.py is what they're supposed to run and be confused when it all passes. Maybe we should explicitly state the command, as we do everywhere else.

Reply:
I think that's a great idea and makes the tutorial more reproducible and easy to follow :)

Reply:
I agree, let's have explicit commands in solutions (otherwise users won't be able to verify, beyond any doubt, that what they did was right and not just coincidentally worked)